zetaclient/chains/bitcoin/observer/mempool.go

```diff
@@ -0,0 +1,277 @@
+package observer
+
+import (
+	"context"
+	"time"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/pkg/errors"
+
+	"github.com/zeta-chain/node/pkg/chains"
+	"github.com/zeta-chain/node/zetaclient/chains/bitcoin/client"
+	"github.com/zeta-chain/node/zetaclient/logs"
+)
+
+const (
+	// PendingTxFeeBumpWaitBlocks is the number of blocks to await before considering a tx stuck in mempool
+	PendingTxFeeBumpWaitBlocks = 3
+
+	// PendingTxFeeBumpWaitBlocksRegnet is the number of blocks to await before considering a tx stuck in mempool in regnet
+	// Note: this is used for E2E test only
+	PendingTxFeeBumpWaitBlocksRegnet = 30
+
+	// blockTimeBTC represents the average time to mine a block in Bitcoin
+	blockTimeBTC = 10 * time.Minute
+)
+
+// LastStuckOutbound contains the last stuck outbound tx information.
+type LastStuckOutbound struct {
+	// Nonce is the nonce of the outbound.
+	Nonce uint64
+
+	// Tx is the original transaction.
+	Tx *btcutil.Tx
+
+	// StuckFor is the duration for which the tx has been stuck.
+	StuckFor time.Duration
+}
+
+// NewLastStuckOutbound creates a new LastStuckOutbound struct.
+func NewLastStuckOutbound(nonce uint64, tx *btcutil.Tx, stuckFor time.Duration) *LastStuckOutbound {
+	return &LastStuckOutbound{
+		Nonce:    nonce,
+		Tx:       tx,
+		StuckFor: stuckFor,
+	}
+}
+
+// PendingTxFinder is a function type for finding the last Bitcoin pending tx.
+type PendingTxFinder func(ctx context.Context, ob *Observer) (*btcutil.Tx, uint64, error)
+
+// StuckTxChecker is a function type for checking if a tx is stuck in the mempool.
+type StuckTxChecker func(ctx context.Context, rpc RPC, txHash string, maxWaitBlocks int64) (bool, time.Duration, error)
+
+// WatchMempoolTxs monitors pending outbound txs in the Bitcoin mempool.
```

Comment:
What watching means? What's the purpose of monitoring here?

Reply:
The purpose of watching:

- Find out the last outbound tx sent to mempool, and there can be many pending txs in mempool; only the last tx (the youngest child) will be tx to be bumped.
- Determine the `stuck` flag. The `stuck == true` flag will lead to a different flow in signer to build an RBF tx, which is different from normal tx building.